File: user-interface-input-controls/app/src/main/java/com/arielmagbanua/userinterface_inputcontrols/MainActivity.java
package com.arielmagbanua.userinterface_inputcontrols;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button buttonsDemo = findViewById(R.id.buttons_demo);
        Button textFieldDemo = findViewById(R.id.text_field_demo);
        Button checkboxDemo = findViewById(R.id.checkbox_demo);
        Button radioButtonDemo = findViewById(R.id.radio_button_demo);

        buttonsDemo.setOnClickListener(this);
        textFieldDemo.setOnClickListener(this);
        checkboxDemo.setOnClickListener(this);
        radioButtonDemo.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.buttons_demo:
                Intent buttonsDemoIntent = new Intent(this, ButtonsDemoActivity.class);
                startActivity(buttonsDemoIntent);
                break;

            case R.id.text_field_demo:
                Intent textFieldDemoIntent = new Intent(this, TextFieldDemoActivity.class);
                startActivity(textFieldDemoIntent);
                break;

            case R.id.checkbox_demo:
                Intent checkboxDemoIntent = new Intent(this, CheckboxDemoActivity.class);
                startActivity(checkboxDemoIntent);
                break;

            case R.id.radio_button_demo:
                Intent radioButtonDemo = new Intent(this, RadioButtonDemoActivity.class);
                startActivity(radioButtonDemo);
                break;
        }
    }
}

File: demo-calculator/app/src/main/java/com/arielmagbanua/democalculator/MainActivity.kt
package com.arielmagbanua.democalculator

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    private lateinit var result: EditText
    private lateinit var newNuber: EditText
    private val displayOperation by lazy(LazyThreadSafetyMode.NONE) {
        findViewById<TextView>(R.id.operation)
    }

    // Variable to hold the operands and
    private var operand1: Double? = null
    private var operand2: Double = 0.0
    private var pendingOperation = "="

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        result = findViewById(R.id.result)
        newNuber = findViewById(R.id.newNumber)

        // Data input buttons
        var button0: Button = findViewById(R.id.button0)
        var button1: Button = findViewById(R.id.button1)
        var button2: Button = findViewById(R.id.button2)
        var button3: Button = findViewById(R.id.button3)
        var button4: Button = findViewById(R.id.button4)
        var button5: Button = findViewById(R.id.button5)
        var button6: Button = findViewById(R.id.button6)
        var button7: Button = findViewById(R.id.button7)
        var button8: Button = findViewById(R.id.button8)
        var button9: Button = findViewById(R.id.button9)
        var buttonDot: Button = findViewById(R.id.buttonDot)

        // Operation buttons
        val buttonEquals = findViewById<Button>(R.id.buttonEquals)
        val buttonDivide = findViewById<Button>(R.id.buttonDivide)
        val buttonMultiply = findViewById<Button>(R.id.buttonMultiply)
        val buttonMinus = findViewById<Button>(R.id.buttonMinus)
        val buttonPlus = findViewById<Button>(R.id.buttonPlus)

        val listener = View.OnClickListener{view ->
            val b = view as Button
            newNuber.append(b.text)
        }

        button0.setOnClickListener(listener)
        button1.setOnClickListener(listener)
        button2.setOnClickListener(listener)
        button3.setOnClickListener(listener)
        button4.setOnClickListener(listener)
        button5.setOnClickListener(listener)
        button6.setOnClickListener(listener)
        button7.setOnClickListener(listener)
        button8.setOnClickListener(listener)
        button9.setOnClickListener(listener)
        buttonDot.setOnClickListener(listener)
    }
}
